File: tests/src/test/java/datomicJava/client/api/async/AsyncTxReportTest.java
package datomicJava.client.api.async;

import datomicJava.SetupAsync;
import org.junit.Test;

import java.util.concurrent.ExecutionException;

import static org.hamcrest.MatcherAssert.assertThat;
import static org.hamcrest.Matchers.is;
import static org.hamcrest.collection.IsEmptyCollection.empty;


public class AsyncTxReportTest extends SetupAsync {

    public AsyncTxReportTest(String name) {
        system = name;
    }

    @Test
    public void txReportOps() throws ExecutionException, InterruptedException {
        assertThat(films(filmDataTx.dbBefore()), is(empty()));
        assertThat(films(filmDataTx.dbAfter()), is(threeFilms));

        // Tx datom + 3 entities * 3 attributes transacted
        assertThat(filmDataTx.txData().count(), is((long) 1 + 3 * 3));

        if (isDevLocal()) {
            // No temp ids created with dev-local setup
            assertThat(filmDataTx.tempIds().size(), is(0));
        } else {
            assertThat(filmDataTx.tempIds().size(), is(3));
        }
    }
}

File: tests/src/test/scala/datomicScala/client/api/async/AsyncTxReportTest.scala
package datomicScala.client.api.async

import datomicScala.SpecAsync


class AsyncTxReportTest extends SpecAsync {
  sequential


  "4 txReport ops" in new AsyncSetup {

    films(filmDataTx.dbBefore) == Nil
    films(filmDataTx.dbAfter) == threeFilms

    // Tx datom + 3 entities * 3 attributes transacted
    filmDataTx.txData.count === 1 + 3 * 3

    if (isDevLocal) {
      // No temp ids created with dev-local setup
      filmDataTx.tempIds.size === 0
    } else {
      filmDataTx.tempIds.size === 3
    }
  }
}